### PR TITLE
[UI] Add MeshModel  Charts in Dashboard 

### LINKED ISSUE #8037: [UI]  Add Treemap and Donut Chart Visuals to Meshery UI Dashboard
#### Current Behavior
Currently, the Meshery UI dashboard lacks  important visualizations that can provide valuable insights. This issue proposes the addition of a Treemap chart and a Donut chart to enhance the dashboard's visual representation.

#### Desired Behavior
Treemap Chart:

- Display MeshModel Categories as boxes, sized according to the count of Models.
- Use visually appealing colors to represent different categories.
- Provide tooltips on hover to show category information.

Donut Chart for MeshModel Constructs:

1. Display proportions of Model, Components, Relationships, and Policies.
2. Use distinct colors for each construct segment.
3. Show tooltips on hover with detailed information.

#### Implementation
- Use Billboard.js for charting . 
- Retrieve data for the charts from the relevant APIs or data sources within Meshery.

#### Mockups

---

#### Contributor [Guides](https://docs.meshery.io/project/contributing) and [Handbook](https://layer5.io/community/handbook)

- 🎨 Wireframes and designs for Meshery UI in [Figma](https://www.figma.com/file/SMP3zxOjZztdOLtgN4dS2W/Meshery-UI)
- 🖥 [Contributing to Meshery UI](https://docs.meshery.io/project/contributing/contributing-ui)
- 🙋🏾🙋🏼 Questions: [Discussion Forum](https://discuss.layer5.io) and [Community Slack](http://slack.layer5.io)

### PR DESCRIPTION
**Pull request by aabidsofi19**
This Pr Adds Two Charts Related To MeshModels  :
- Pie Chart for models per category 
- Donut Chart for MeshModel Contructs 

**Notes for Reviewers**
-  Bumped BillBoardJs to 3.9.0
-  Custom Colors for Charts go round robin 
-  Used Dark Theme for Billboardjs as it works good our theme implementations both light and dark
-  extracted reusable logic into utils and components .

This PR fixes #8037  

image
image
image

**[Signed commits](https://github.com/meshery/meshery/blob/master/CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [x] Yes, I signed my commits.

### CODE REVIEW

**Comment by leecalcote** [comment]
@aabidsofi19 while hovering on the Constructs chart:
Screenshot 2023-07-08 at 7 15 08 PM

**Comment by aabidsofi19** [comment]
> @aabidsofi19 while hovering on the Constructs chart:
> 
> Screenshot 2023-07-08 at 7 15 08 PM
> Note, that in my deployment, I'm experiencing trouble with web sockets (graphql subscriptions).

i just checked again , the hover tooltips are working fine for both graphs 

-  image
-  image

File: ui/components/DashboardMeshModelGraph.js [hunk 1]
```diff
@@ -1,48 +1,56 @@
...
+    arc: {
+      cornerRadius: {
+        ratio: 0.1
+      }
+    },
+    bindto: "#donutCornerRadius",
+    donut: {
+      title: "Capabilities\nRegistry"
+    },
+    bindto: "#multilineTitle",
```
Comment (aabidsofi19) [inline ui/components/DashboardMeshModelGraph.js:48]: we are  having duplicate bindto .

**Comment by aabidsofi19** [comment]
@leecalcote  there are no endpoints for fetching policies in api docs  and  also no section  in the settings .

File: ui/components/DashboardMeshModelGraph.js (hunk 1 above)
Comment (leecalcote) [reply ui/components/DashboardMeshModelGraph.js]: Thanks for reminding us

**Comment by leecalcote** [comment]
> @leecalcote there are no endpoints for fetching policies in api docs and also no section in the settings .

Hence, the TODO.

**Comment by leecalcote** [comment]
> still i dont think it should be related to graphql connection as we not fetching anything on hover .

I'm sure that you're right.

**Review by leecalcote — APPROVED** [review]
There's some good work in here, @aabidsofi19. 👏

**Comment by leecalcote** [comment]
You just changed the face of the project with this dashboard update, @aabidsofi19. 🥇

**Pull request merged**